java/src/Ice/src/main/java/com/zeroc/Ice/WSEndpoint.java

```diff
@@ -172,30 +172,25 @@ public WSEndpoint endpoint(EndpointI delEndp) {
     }
 
     @Override
-    public java.util.List<EndpointI> expandIfWildcard() {
-        java.util.List<EndpointI> endps = _delegate.expandIfWildcard();
-        java.util.List<EndpointI> l = new java.util.ArrayList<>();
-        for (EndpointI e : endps) {
-            l.add(e == _delegate ? this : new WSEndpoint(_instance, e, _resource));
+    public java.util.List<EndpointI> expandHost() {
+        java.util.List<EndpointI> underlying = _delegate.expandHost();
+
+        var result = new java.util.ArrayList<EndpointI>(underlying.size());
+
+        for (EndpointI e : underlying) {
+            result.add(endpoint(e));
         }
```

Comment:
I'm surprised there's no way to "map" one into the other.

Reply:
Can't we do?

```
_delegate.expandHost().stream().map(this::endpoint).collect(Collectors.toList());
```

Reply:
Sure, I'd meant that there wasn't a nice way to do this directly on the list, like `Select` in C#.
You have to convert back-and-forth from a `Stream` to a `List`, which is kind of lame.